pkg/pillar/agentlog/agentlog.go

```diff
@@ -396,3 +396,24 @@ func CheckMaxTime(agentName string, start time.Time) {
 			fn, line, agentName, elapsed/time.Second)
 	}
 }
+
+// CheckMaxTimeTopic verifies if the time for a call has exeeded a reasonable
+// number.
+func CheckMaxTimeTopic(agentName string, topic string, start time.Time) {
+	errTime := errorTime
+	warnTime := warningTime
+	if agentName == "nim" {
+		errTime = errorTimeNim
+		warnTime = warningTimeNim
+	}
```

Comment:
Why is there nim hardcoding in agent.log?? Can we make it more modular? Let the agents pass error / warning times ??

Reply:
This code didn't change.
Note that I need two copies of the two flavors of the function since it uses the caller.

Reply:
> Note that I need two copies of the two flavors of the function since it uses the caller

Yup, makes sense.

Reply:
I agree with @kalyan-nidumolu ; we should remove the hard-coded custom timing. If we need a special case, we can have a map with them, and it can look up the times in a map with a default. We probably should avoid hardcoding  a specific agent in here.

Reply:
That's fine, however this PR is to fix a regression caused a recent PR: https://github.com/lf-edge/eve/pull/440
I can instead back out that PR if you prefer. But I think you have more important things on your plate.

Reply:
Yeah fair enough.

Reply:
I've removed this and moved the values out into the agents.